src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -35,6 +40,64 @@ public IndexModel createIndex(CreateIndexRequest createIndexRequest) throws Pine
         return indexModel;
     }
 
+    public <E extends Enum<E>> boolean validateEnums(String stringToValidate, List<E> validEnums) {
+        return validEnums.stream()
+                .anyMatch(validEnum -> Objects.equals(stringToValidate, validEnum.toString()));
+    }
+
+    public IndexModel createServerlessIndex(String indexName, String metric, int dimension, String cloud,
+                                            String region) {
+        if (indexName == null || indexName.isEmpty()) {
+            throw new PineconeValidationException("Index name cannot be null or empty");
+        }
+
+        if (metric == null || metric.isEmpty()) {
+            throw new PineconeValidationException("Metric cannot be null or empty. Must be 'euclidean', 'cosine' or 'dot-product'");
+        }
+        List<IndexMetric> indexMetricEnums = Arrays.asList(IndexMetric.values());
+        if (!validateEnums(metric, indexMetricEnums)) {
+            throw new PineconeValidationException(String.format("Metric must be one of %s", IndexMetric.values()));
+        }
+
+        if (dimension < 1) {
+            throw new PineconeValidationException("Dimension must be greater than 0");
+        }
```

Comment:
Do we have an upper bound on dimension? I am not sure what we have in our other SDKs.

Reply:
Technically [the limit is 4.2B](https://docs.pinecone.io/reference/limits) haha (for dense vectors). I don't see in the Python or Node SDKs, at least any upper limit enacted through code.

I'll add a link to our limits section, though. That'll be helpful for users.

Reply:
https://github.com/pinecone-io/pinecone-java-client/pull/94/commits/bc59fc962bc67f19771d71b4802aef94131e2f78